Dockerfile

```diff
@@ -1,103 +1,137 @@
 
-# install system dependencies
-RUN apt update -qq && apt install -qq --yes --no-install-recommends \
-    python virtualenv python-pip wget curl lbzip2 patch bsdtar sudo && \
-    rm -rf /var/lib/apt/lists/*
+# download and install Android NDK
+RUN curl --location --progress-bar --insecure \
+        "${ANDROID_NDK_DL_URL}" \
+        --output "${ANDROID_NDK_ARCHIVE}" \
+    && mkdir --parents "${ANDROID_NDK_HOME_V}" \
+    && unzip -q "${ANDROID_NDK_ARCHIVE}" -d "${ANDROID_HOME}" \
+    && ln -sfn "${ANDROID_NDK_HOME_V}" "${ANDROID_NDK_HOME}" \
+    && rm -rf "${ANDROID_NDK_ARCHIVE}"
 
-# build dependencies
-# https://buildozer.readthedocs.io/en/latest/installation.html#android-on-ubuntu-16-04-64bit
-RUN dpkg --add-architecture i386 && apt update -qq && apt install -qq --yes --no-install-recommends \
-    build-essential ccache git libncurses5:i386 libstdc++6:i386 libgtk2.0-0:i386 \
-    libpangox-1.0-0:i386 libpangoxft-1.0-0:i386 libidn11:i386 python2.7 python2.7-dev \
-    openjdk-8-jdk zip unzip zlib1g-dev zlib1g:i386
 
-# specific recipes dependencies (e.g. libffi requires autoreconf binary)
-RUN apt install -qq --yes --no-install-recommends \
-    autoconf automake cmake gettext libltdl-dev libtool pkg-config && \
-    rm -rf /var/lib/apt/lists/*
+ENV ANDROID_SDK_HOME="${ANDROID_HOME}/android-sdk"
 
-# download and install Android NDK
-RUN curl --location --progress-bar "${ANDROID_NDK_DL_URL}" --output "${ANDROID_NDK_ARCHIVE}" && \
-    mkdir --parents "${ANDROID_NDK_HOME_V}" && \
-    unzip -q "${ANDROID_NDK_ARCHIVE}" -d "${ANDROID_HOME}" && \
-    ln -sfn "${ANDROID_NDK_HOME_V}" "${ANDROID_NDK_HOME}" && \
-    rm -rf "${ANDROID_NDK_ARCHIVE}"
-
-# download and install CrystaX NDK
-# added `gnutls_handshake` flag to workaround random `gnutls_handshake()` issues
-RUN curl --location --progress-bar "${CRYSTAX_NDK_DL_URL}" --output "${CRYSTAX_NDK_ARCHIVE}" --insecure && \
-    bsdtar -xf "${CRYSTAX_NDK_ARCHIVE}" --directory "${ANDROID_HOME}" \
-    --exclude=crystax-ndk-${CRYSTAX_NDK_VERSION}/docs \
-    --exclude=crystax-ndk-${CRYSTAX_NDK_VERSION}/samples \
-    --exclude=crystax-ndk-${CRYSTAX_NDK_VERSION}/tests \
-    --exclude=crystax-ndk-${CRYSTAX_NDK_VERSION}/toolchains/renderscript \
-    --exclude=crystax-ndk-${CRYSTAX_NDK_VERSION}/toolchains/x86_64-* \
-    --exclude=crystax-ndk-${CRYSTAX_NDK_VERSION}/toolchains/llvm-* \
-    --exclude=crystax-ndk-${CRYSTAX_NDK_VERSION}/toolchains/aarch64-* \
-    --exclude=crystax-ndk-${CRYSTAX_NDK_VERSION}/toolchains/mips64el-* && \
-    ln -sfn "${CRYSTAX_NDK_HOME_V}" "${CRYSTAX_NDK_HOME}" && \
-    rm -rf "${CRYSTAX_NDK_ARCHIVE}"
+# get the latest version from https://developer.android.com/studio/index.html
+ENV ANDROID_SDK_TOOLS_VERSION="3859397"
+ENV ANDROID_SDK_TOOLS_ARCHIVE="sdk-tools-linux-${ANDROID_SDK_TOOLS_VERSION}.zip"
+ENV ANDROID_SDK_TOOLS_DL_URL="https://dl.google.com/android/repository/${ANDROID_SDK_TOOLS_ARCHIVE}"
```

Comment:
Why don't we keep all the env configuration top most? Is it for layer caching reasons? For me the idea is if it's on top we can easily see what can be "configured"